core/src/com/unciv/models/metadata/GameSettings.kt

```diff
@@ -68,4 +74,64 @@ class GameSettings {
         if (tutorialTasksCompleted.add(tutorialTask))
             save()
     }
+
+    fun updateLocaleFromLanguage() {
+        val bannedCharacters = listOf(' ', '_', '-', '(', ')') // Things not to have in enum names
+        val languageName = language.filterNot { it in bannedCharacters }
+        val code = try {
+            LocaleCode.valueOf(languageName)
+        } catch (e: IllegalArgumentException) {
+            LocaleCode.English
+        }
+        locale = Locale(code.language, code.country)
+    }
+
+    fun getCurrentLocale(): Locale {
+        if (locale == null)
+            updateLocaleFromLanguage()
+        return locale!!
+    }
+}
+
+enum class LocaleCode(var language: String, var country: String) {
+    Arabic("ar", "IQ"),
+    BrazilianPortuguese("pt", "BR"),
+    Bulgarian("bg", "BG"),
+    Catalan("ca", "ES"),
+    Croatian("hr", "HR"),
+    Czech("cs", "CZ"),
+    Danish("da", "DK"),
+    Dutch("nl", "NL"),
+    English("en", "US"),
+    Estonian("et", "EE"),
+    Finnish("fi", "FI"),
+    French("fr", "FR"),
+    German("de", "DE"),
+    Greek("el", "GR"),
+    Hindi("hi", "IN"),
+    Hungarian("hu", "HU"),
+    Indonesian("in", "ID"),
+    Italian("it", "IT"),
+    Japanese("ja", "JP"),
+    Korean("ko", "KR"),
+    Latvian("lv", "LV"),
+    Lithuanian("lt", "LT"),
+    Malay("ms", "MY"),
+    Norwegian("no", "NO"),
```

Comment:
I wonder if we ever are lucky enough to get a norwegian xlator if they wouldn't prefer their files to be named just Bokmål and Nynorsk - nah, no reason to worry now :laughing:

Reply:
Mainly I wanted to avoid having a non-ASCII character in the name. If a Norwegian ever shows up we can ask their opinion.